src/connection/mod.rs

```diff
@@ -65,7 +63,7 @@ impl Debug for Connection {
 }
 
 impl Connection {
-    async fn connect(server_list: &ServerList) -> Result<Self, ConnectionError> {
+    pub async fn connect(server_list: &ServerList) -> Result<Self, ConnectionError> {
```

Comment:
I would prefer not allowing users to have access to a `Connection` that isn't authenticated yet to make it impossible for a user to try and send requests to a non-authenticated `Connection`.

An option would be to have an `UnauthenticatedConnection` struct that wraps the `Connection`. This would then be passed to the various auth methods and has the `login_with_token` method to get the full, authenticated `Connection`

Reply:
I've implemented the `UnAuthenticationConnection` in 6839fe1e2033bf28eddb097e11e593dcdc782679